pkg/globalinit/plugin_discovery_source_update.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package globalinit
+
+import (
+	"fmt"
+	"io"
+
+	cliconfig "github.com/vmware-tanzu/tanzu-cli/pkg/config"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/lastversion"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+)
+
+// This global initializer checks if the last executed CLI version is < 1.3.0.
+// If so it tries to update the plugin discovery source if needed
+
+func init() {
+	RegisterInitializer("Plugin Discovery Source Updater", triggerForPluginDiscoverySourceUpdater, updatePluginDiscoverySource)
+}
+
+func triggerForPluginDiscoverySourceUpdater() bool {
+	// If the last executed CLI version is < 1.3.0, we need to update the discovery image source
+	return lastversion.GetLastExecutedCLIVersion() == lastversion.OlderThan1_3_0
+}
+
+// updatePluginDiscoverySource updates the plugin discovery source to point CLI to
+// new `projects.packages.broadcom.com` and also clears the EULA acceptance state
+// stored in the CLI config file. So that EULA prompt is shown to the user again.
+func updatePluginDiscoverySource(outStream io.Writer) error {
+	ds, err := config.GetCLIDiscoverySource(cliconfig.DefaultStandaloneDiscoveryName)
+	if err != nil {
+		return err
+	}
+
+	if ds == nil || ds.OCI == nil || ds.OCI.Image != constants.TanzuCLIOldCentralPluginDiscoveryImage {
+		// User must have manually updated discovery source. So do not overwrite that change.
+		return nil
+	}
+
+	// Considering we are updating the discovery source to point to new registry,
+	// User must be prompted to access the EULA again. So, let's first unset EULA status
+	err = config.SetEULAStatus(config.EULAStatusUnset)
```

Comment:
I have always expected the situation when we need to reprompt EULA involves (1) setting a new major.minor 
in CurrentEULAVersion, and (2) updating the eulaPromptMsg, instead of explicitly unsetting the status.

Can we consider this a special situation that doesn't follow this pattern?
Could we have just done (1) instead?

Reply:
Good point. I didn't think about (1) at all before you mentioned it. We can go with approach (1) as well.

The minor difference will be that with the current approach, we do not prompt EULA for the users using the custom registry. But with the approach (1) you mentioned, it will always be prompted for the users. I think it makes sense to prompt the EULA again for all the users as users can use the `tanzu plugin download-bundle` which will download images from a new location by default.

So, I am inclined towards updating this PR with approach (1) of updating `CurrentEULAVersion` with a new major.minor.